python-threatexchange/threatexchange/TE.py

```diff
@@ -13,15 +13,77 @@
 # General Python dependencies
 import os
 import re
-import urllib
-import urllib.error
+import requests
+
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
 import urllib.parse
-import urllib.request
+
+
+DEFAULT_TIMEOUT = 5  # seconds
+
+
+class TimeoutHTTPAdapter(HTTPAdapter):
+    """
+    Plug into requests to get a well-behaved session that does not wait for eternity.
+    H/T: https://findwork.dev/blog/advanced-usage-python-requests-timeouts-retries-hooks/#setting-default-timeouts
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.timeout = DEFAULT_TIMEOUT
+        if "timeout" in kwargs:
+            self.timeout = kwargs["timeout"]
+            del kwargs["timeout"]
+        super().__init__(*args, **kwargs)
+
+    def send(self, request, **kwargs):
+        timeout = kwargs.get("timeout")
+        if timeout is None:
+            kwargs["timeout"] = self.timeout
+        return super().send(request, **kwargs)
+
+
+DEFAULT_TE_BASE_URL = "https://graph.facebook.com/v6.0"
+retry_strategy = Retry(
+    total=4,
+    status_forcelist=[429, 500, 502, 503, 504],
+    method_whitelist=["HEAD", "GET", "OPTIONS"],
+)
+adapter = TimeoutHTTPAdapter(timeout=60, max_retries=retry_strategy)
```

Comment:
Nit, not blocking: would prefer this also be scoped to the get_fb_graph_api function to prevent from adding things to the global namespace, or at the very least prefixed with underscore (_) to indicate that it's part of the implementation detail.

Reply:
This was done.